src/components/content/order/locks/Locks.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import React, { useRef, useState } from 'react';
+import { Button, Form, Space, Switch } from 'antd';
+import '../../../../styles/locks.css';
+import {
+    ApiError,
+    DeployedServiceDetails,
+    Response,
+    ServiceLockConfig,
+    ServiceService,
+    VendorHostedDeployedServiceDetails,
+} from '../../../../xpanse-api/generated';
+import { useMutation } from '@tanstack/react-query';
+
+export const Locks = ({
+    currentSelectedService,
+}: {
+    currentSelectedService: DeployedServiceDetails | VendorHostedDeployedServiceDetails;
+}): React.JSX.Element => {
+    const [form] = Form.useForm();
+    const requestResult = useRef<string[]>([]);
+    const serviceLockConfig = useRef<{ id: string; lockConfig: ServiceLockConfig } | undefined>(undefined);
+    const [destroyChecked, setDestroyChecked] = useState<boolean>(
```

Comment:
we can avoid storing this data again in the state. All data in the API response can be directly used.